Incorrectly written test cases masking test failure

**Bug** 

I noticed an unhandled promise rejection somewhere in our test suite, which made me look a bit deeper.  Several of our test cases are written wrong and one of them is actually failing, but because the way it is written is causing it to look like it is passing.

The unhandled promise occurs [in this block of code](https://github.com/dojo/cli-build/blob/master/tests/unit/plugins/InjectModulesPlugin.ts#L375-L381):

```ts
return new Promise((resolve, reject) => {
	resolver({ resource: '/test/module.js' }, () => {
		resolve();
		assert.isTrue((<any> plugin.resolve).called);
		assert.isTrue((<any> plugin.createModules).calledWith([ { context: '/test', request: './module' } ]));
	});
});
```

First, once a promise is resolved, no other tracking by Intern occurs.  Second, throws in a `Promise` callback do not automatically reject the promise.  If we wanted this code to actually _fail_ when the assertions throw, it would need to look like this:

```ts
return new Promise((resolve, reject) => {
	resolver({ resource: '/test/module.js' }, () => {
		try {
			assert.isTrue((<any> plugin.resolve).called);
			assert.isTrue((<any> plugin.createModules).calledWith([ { context: '/test', request: './module' } ]));
		}
		catch (e) {
			reject(e);
		}
		resolve();
	});
});
```

While this one appears to be the only one failing, there are several others that are written in a way they won't actually _fail_ irrespective of the assertions succeeding or failing.

This also means that there is either a defect with the test case or the code, because if changed, you get the following error:

```
× inject-modules - "resolver" factory plugin - should generate data for matching issuers (0.003s)
AssertionError: expected false to be true
  at Assertion.<anonymous>  <node_modules/intern/browser_modules/chai/chai.js:567:10>
  at Assertion.addProperty  <node_modules/intern/browser_modules/chai/chai.js:4240:29>
  at Function.assert.isTrue  <node_modules/intern/browser_modules/chai/chai.js:2483:31>
  at resolver  <tests/unit/plugins/InjectModulesPlugin.ts:379:13>
  at resolve.then.then  </Users/kitsonk/github/cli-build/src/plugins/InjectModulesPlugin.ts:167:9>
  at runMicrotasksCallback  <internal/process/next_tick.js:64:5>
  at _combinedTickCallback  <internal/process/next_tick.js:73:7>
  at process._tickCallback  <internal/process/next_tick.js:104:9>
```

--- a/tests/unit/plugins/InjectModulesPlugin.ts
+++ b/tests/unit/plugins/InjectModulesPlugin.ts
@@ -374,9 +374,14 @@ describe('inject-modules', () => {
 
 			return new Promise((resolve, reject) => {
 				resolver({ resource: '/test/module.js' }, () => {
-					resolve();
-					assert.isTrue((<any> plugin.resolve).called);
-					assert.isTrue((<any> plugin.createModules).calledWith([ { context: '/test', request: './module' } ]));
+					try {
+						assert.isTrue((<any> plugin.resolve).called);
+						assert.isTrue((<any> plugin.createModules).calledWith([ { contextInfo: {}, context: '/test', request: './module' } ]));
+						resolve();
+					}
+					catch (error) {
+						reject(error);
+					}
 				});
 			});
 		});
@@ -391,8 +396,13 @@ describe('inject-modules', () => {
 					(<any> plugin.resolve).restore();
 					sinon.spy(plugin, 'resolve');
 					resolver({ resource: '/test/module.js' }, () => {
-						resolve();
-						assert.isFalse((<any> plugin.resolve).called);
+						try {
+							assert.isFalse((<any> plugin.resolve).called);
+							resolve();
+						}
+						catch (error) {
+							reject(error);
+						}
 					});
 				});
 			});
@@ -405,10 +415,15 @@ describe('inject-modules', () => {
 
 			return new Promise((resolve, reject) => {
 				resolver({ resource: '/test/module.js' }, (error: Error) => {
-					resolve();
-					assert.strictEqual(error.message, 'mock error');
-					assert.isTrue((<any> plugin.resolve).called);
-					assert.isFalse((<any> plugin.createModules).called);
+					try {
+						assert.strictEqual(error.message, 'mock error');
+						assert.isTrue((<any> plugin.resolve).called);
+						assert.isFalse((<any> plugin.createModules).called);
+						resolve();
+					}
+					catch (error) {
+						reject(error);
+					}
 				});
 			});
 		});

